torchgeo/datasets/digital_typhoon.py

```diff
@@ -0,0 +1,454 @@
+from typing import Any, ClassVar, TypedDict
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_url, lazy_import, percentile_normalization
+
+
+class _SampleSequenceDict(TypedDict):
+    """Sample sequence dictionary."""
+
+    id: str
+    seq_id: list[int]
+
+
+class DigitalTyphoonAnalysis(NonGeoDataset):
+    """Digital Typhoon Dataset for Analysis Task.
+
+    This dataset contains typhoon-centered images, derived from hourly infrared channel
+    images captured by meteorological satellites. It incorporates data from multiple
+    generations of the Himawari weather satellite, dating back to 1978. These images
+    have been transformed into brightness temperatures and adjusted for varying
+    satellite sensor readings, yielding a consistent spatio-temporal dataset that
+    covers over four decades.
+
+    See `the Digital Typhoon website
+    <http://agora.ex.nii.ac.jp/digital-typhoon/dataset/>`_
+    for more information about the dataset.
+
+    Dataset features:
+
+    * infrared channel images from the Himawari weather satellite (512x512 px)
+      at 5km spatial resolution
+    * auxiliary features such as wind speed, pressure, and more that can be used
+      for regression or classification tasks
+    * 1,099 typhoons and 189,364 images
+
+    Dataset format:
+
+    * hdf5 files containing the infrared channel images
+    * .csv files containing the metadata for each image
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.20783/DIAS.664
+
+    .. versionadded:: 0.6
+    """
+
+    valid_tasks: ClassVar[list[str]] = ['classification', 'regression']
+    aux_file_name: ClassVar[str] = 'aux_data.csv'
+
+    valid_features: ClassVar[list[str]] = [
+        'year',
+        'month',
+        'day',
+        'hour',
+        'grade',
+        'lat',
+        'lng',
+        'pressure',
+        'wind',
+        'dir50',
+        'long50',
+        'short50',
+        'dir30',
+        'long30',
+        'short30',
+        'landfall',
+        'intp',
+    ]
+
+    url: ClassVar[str] = (
+        'https://huggingface.co/datasets/torchgeo/digital_typhoon/resolve/main/WP.tar.gz{0}'
+    )
+
+    md5sums: ClassVar[dict[str, str]] = {
+        'aa': '3af98052aed17e0ddb1e94caca2582e2',
+        'ab': '2c5d25455ac8aef1de33fe6456ab2c8d',
+    }
+
+    min_input_clamp = 170.0
+    max_input_clamp = 300.0
+
+    data_root = 'WP'
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        task: str = 'regression',
+        features: Sequence[str] = ['wind'],
+        targets: list[str] = ['wind'],
+        sequence_length: int = 3,
+        min_feature_value: dict[str, float] | None = None,
+        max_feature_value: dict[str, float] | None = None,
```

Comment:
I wonder if these should have defaults, or if that should be left up to the user. Do the dataset authors use or prescribe any specific ranges?

Reply:
By default all values are included.

Reply:
Yes, I'm wondering if we should have defaults that strip out, e.g., negative wind speeds.